apertium_apy/handlers/translate.py

```diff
@@ -150,33 +150,36 @@ def get_format(self):
 
         return deformat, reformat
 
-    @gen.coroutine
-    def translate_and_respond(self, pair, pipeline, to_translate, mark_unknown, nosplit=False, deformat=True, reformat=True):
-        mark_unknown = mark_unknown in ['yes', 'true', '1']
-        self.note_pair_usage(pair)
-        before = self.log_before_translation()
-        translated = yield pipeline.translate(to_translate, nosplit, deformat, reformat)
-        self.log_after_translation(before, len(to_translate))
-        self.send_response({
-            'responseData': {
-                'translatedText': self.maybe_strip_marks(mark_unknown, pair, translated),
-            },
-            'responseDetails': None,
-            'responseStatus': 200,
-        })
-        self.clean_pairs()
-
     @gen.coroutine
     def get(self):
-        pair = self.get_pair_or_error(self.get_argument('langpair'),
-                                      len(self.get_argument('q')))
+        query_list = self.get_arguments('q')
+        langpair = self.get_argument('langpair')
+        translation_futures = []
+        response = []
+        pair = self.get_pair_or_error(langpair, len(''.join(query_list)))
         if pair is not None:
+            self.note_pair_usage(pair)
             pipeline = self.get_pipeline(pair)
             deformat, reformat = self.get_format()
-            yield self.translate_and_respond(pair,
-                                             pipeline,
-                                             self.get_argument('q'),
-                                             self.get_argument('markUnknown', default='yes'),
-                                             nosplit=False,
-                                             deformat=deformat,
-                                             reformat=reformat)
+            mark_unknown = self.get_argument('markUnknown', default='yes') in ['yes', 'true', '1']
+            for query in query_list:
+                before = self.log_before_translation()
+                translation_futures.append(pipeline.translate(query, nosplit=False, deformat=deformat, reformat=reformat))
+                self.log_after_translation(before, len(query))
+
+        translations = yield gen.multi_future(translation_futures)
```

Comment:
I think this needs to be indented. 

```
if pair is not None:
    return
```

is also fine. I like early returns but it's really up to you.

Reply:
Just to clear something you mean
```
if pair is None:
    return 
```
right?

Reply:
Ah, yeah. My bad.

Reply:
Fine, I'll do it right now.